rust/connlib/libs/tunnel/src/device_channel_unix.rs

```diff
@@ -60,8 +60,8 @@ impl DeviceChannel {
     }
 }
 
-pub(crate) async fn create_iface() -> Result<(IfaceConfig, DeviceChannel)> {
-    let dev = Arc::new(IfaceDevice::new().await?.set_non_blocking()?);
+pub(crate) async fn create_iface(fd: Option<i32>) -> Result<(IfaceConfig, DeviceChannel)> {
+    let dev = Arc::new(IfaceDevice::new(fd).await?.set_non_blocking()?);
```

Comment:
```suggestion
    let dev = Arc::new(unsafe { IfaceDevice::new(fd).await?.set_non_blocking()? });
```

Fixes CI build issue
```rust
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> connlib/libs/tunnel/src/device_channel_unix.rs:64:24
   |
64 |     let dev = Arc::new(IfaceDevice::new(fd).await?.set_non_blocking()?);
   |                        ^^^^^^^^^^^^^^^^^^^^ call to unsafe function
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior
```

Reply:
Fixed by removing`unsafe` from the function definition.